### PR TITLE
update ejabberd to  v21.01

### LINKED ISSUE #2095: Upgrade EJABBERD package from version 2 to the current version which is 16.01
The ejabberd package is version 2, but the current version is 16. 
I can only imagine how many security related fixes there are between those two versions. How do we upgrade that? Is there something the general community can help with?

Thanks
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30134755-upgrade-ejabberd-package-from-version-2-to-the-current-version-which-is-16-01?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2260: Ejabberd not starting after installation
Hello, 
Would like to report that Ejabberd fails to run after new installation on Synology.
No error msg during installation. However after install, the app attempt to start for 2 minutes after failing with the msg (Failed to run the package service)

System is: DS412+
OS version: DSM 6.0-7321 Update 1

Your help in this matter is appreciated

Thank you.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32948264-ejabberd-not-starting-after-installation?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2397: ejabberd compiling error
using erlang 19 and ejabberd 16 i get the folowing error when i run make

=> ejabberd (compile)

=ERROR REPORT==== 4-Jul-2016::09:31:32 ===
Error in process <0.36.0> with exit value:
{{case_clause,[]},
 [{code_server,split_base,1,[{file,"code_server.erl"},{line,639}]},
  {code_server,archive_subdirs,1,[{file,"code_server.erl"},{line,825}]},
  {code_server,do_insert_name,3,[{file,"code_server.erl"},{line,819}]},
  {code_server,do_add,4,[{file,"code_server.erl"},{line,739}]},
  {code_server,add_path,4,[{file,"code_server.erl"},{line,714}]},
  {code_server,handle_call,3,[{file,"code_server.erl"},{line,255}]},
  {code_server,loop,1,[{file,"code_server.erl"},{line,150}]}]}
2016-07-04 09:31:33 std_info  
    application: kernel
    exited: shutdown
    type: permanent
{"Kernel pid terminated",application_controller,"{application_terminated,kernel,shutdown}"}

Crash dump is being written to: erl_crash.dump...done
Kernel pid terminated (application_controller) ({application_terminated,kernel,shutdown})

any solutions 
### Setup

_Package Name:_
_Package Version:_

_NAS Model:_
_NAS Architecture:_
_DSM version:_
### Expected behavior

Tell us what should happen
### Actual behavior

Tell us what happens instead
### Steps to reproduce

_1._
_2._
_3._
#### Package log

_Check Package Center or `/usr/local/{package}/var/`_

```
Insert the package log here
```
#### Other logs

_E.g. `/var/log/messages` or `/var/log/synopkg.log`_

```
Insert log here
```

### LINKED ISSUE #3770: erlang compilation is failing
### Setup
_Package Name:_ Erlang
_Package Version:_ R16B03-1

_NAS Model:_  DS 716+II
_NAS Architecture:_ braswell
_DSM version:_ 6.1

### Expected behavior
The Erlang compilation in the cross directory should compile fine

### Actual behavior
The error appear below : 
In file included from beam/erl_threads.h:267:0,
                 from beam/erl_smp.h:27,
                 from beam/sys.h:416,
                 from hipe/hipe_bif0.c:27:
hipe/hipe_amd64_primops.h:21:13: erreur: «am_inc_stack_0» non déclaré ici (hors de toute fonction)
 PRIMOP_LIST(am_inc_stack_0, &nbif_inc_stack_0)
             ^

### Steps to reproduce
_1._ In the Spksrc directory : make setup
_2._ cd native/erlang
_3._ make
_2._ cd cross/erlang
_3._ make arch-braswell

### PR DESCRIPTION
**Pull request by hgy59**
_Motivation:_  Ejabberd does not build since some time
_Linked issues:_  #4214. closes #3770, closes #2397, closes #2260, closes #2095

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Remarks
- update erlang to v23.2.5
- update ejabberd to v21.01
- ppc853x is not supported anymore
- use generic service setup
- activate install wizard

### CODE REVIEW

**Review by ymartin59 — APPROVED** [review]

**Comment by willemb2** [comment]
@hgy59 On my DS216+II with DSM 6.2.3-25426 Update 3 the Package Center - Community still shows ejabberd 2.1.13-1. But I don't know what the route is between GitHub and the Package Center, so maybe I'm too impatient.

**Comment by hgy59** [comment]
> 
> 
> @hgy59 On my DS216+II with DSM 6.2.3-25426 Update 3 the Package Center - Community still shows ejabberd 2.1.13-1. But I don't know what the route is between GitHub and the Package Center, so maybe I'm too impatient.

There is no automatic update of the synocommunity Package Center from github builds.
And there are no updates from Pull-Requests. The Package Center is updated manually by maintainers after a PR is merged back to the master branch. Before upload to the package center such builds are verified, and after upload it takes up to 48 hours until the package is visible in the DSM package center.

For manual installation you can use the build artifacts of successful github build actions on each PR.
   - click `Details` on one of the build tasks under `All checks have passed`
   - Open the popup menu under `Artifacts`
   - Click `packages`

This will download `packages.zip` containing all successfully built packages.

The packages containing x64-6.1, aarch64-6.1 and armv7-6.1 in the name are builds for generic architectures. You will find the matching generic (or regular) arch for your device on the wiki page [Package Archs](https://github.com/SynoCommunity/spksrc/wiki/Synology-and-SynoCommunity-Package-Architectures).
Extract the spk matching your Diskstation Model and use it for manual installation.

_HINT: Version 6.1 is compatible with DSM 6.1, 6.2, ... 6.2.3 (but not for DSM 7+)_

**Comment by hgy59** [comment]
A dedicated erlang package is on the way with #4437. This should be used here too.

**Review by ymartin59 — APPROVED** [review]

**Pull request merged**
